exporter/exporterhelper/internal/persistent_storage_test.go

```diff
@@ -174,6 +172,7 @@ func TestPersistentStorage_CorruptedData(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
+			ext := createStorageExtension(path)
```

Comment:
Why this change? I didn't see it mentioned in the description.

Reply:
This essentially moves [this line](https://github.com/open-telemetry/opentelemetry-collector/pull/4592/files#diff-b70537f6db93c86aa3901865578263fe3dd9557c75db043e585de2983135913aL114) here, which makes sure that each run gets a new storage extension instance